lib/new_relic/agent/instrumentation/active_support_logger.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative 'active_support_logger/instrumentation'
+require_relative 'active_support_logger/chain'
+require_relative 'active_support_logger/prepend'
+
+DependencyDetection.defer do
+  named :active_support_logger
+
+  depends_on { defined?(::ActiveSupport::Logger) }
+
+  executes do
+    ::NewRelic::Agent.logger.info 'Installing ActiveSupport::Logger instrumentation'
```

Comment:
From what I understand, these `execute` blocks ultimate result in an array of separate executable blocks that are fired later. I could see how it provides value to keep separate independent chunks of logic in different blocks. In this case though with a single logger statement describing what the subsequent `executes` block is doing, I think it would make more sense to use a single `executes` block and have the logger statement live with the code it's describing.

Reply:
Yeah, I think that seems reasonable. I looked through a few files and found that we do it both ways in different places, so I think either one would be consistent with other code.